Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -0,0 +1,862 @@
+
+
+def attachment_upload_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """Upload attachment to certain incident in TOPdesk"""
+    file_entry = args.get('file', None)
+    if not file_entry:
+        raise Exception("Found no file entry.")
+
+    file_name = demisto.dt(demisto.context(), "File(val.EntryID=='" + file_entry + "').Name")
+    if not file_name:  # in case of info file
+        file_name = demisto.dt(demisto.context(), "InfoFile(val.EntryID=='" + file_entry + "').Name")
+    if not file_name:
+        raise ValueError(f"Could not fine file in entry with entry_id: {file_entry}")
+    file_name = file_name[0] if isinstance(file_name, list) else file_name  # If few files
+
+    invisible_for_caller_str = args.get('invisible_for_caller', None)
+    invisible_for_caller = None
+    if invisible_for_caller_str:
+        if invisible_for_caller_str.lower() in ['true', 't', 'yes', 'y', '1']:
+            invisible_for_caller = True
+        else:
+            invisible_for_caller = False
+
+    response = client.attachment_upload(incident_id=args.get('id', None),
+                                        incident_number=args.get('number', None),
+                                        file_entry=file_entry,
+                                        file_name=file_name,
+                                        invisible_for_caller=invisible_for_caller,
+                                        file_description=args.get('file_description', None))
+
+    if not response.get("downloadUrl", None):
+        raise Exception(f"Failed uploading file: {response}")
+
+    headers = ['id', 'file name', 'download url', 'size', 'description',
+               'invisible for caller', 'entry date', 'operator']
+    half_camelized_headers = [half_camelize(header, ' ') for header in headers]
+    readable_attachment: List[Dict[str, Any]] = [{}]
+    for header, half_camelized_header in zip(headers, half_camelized_headers):
+        readable_attachment[0][header] = response.get(half_camelized_header, None)
+
+    readable_output = tableToMarkdown(f"{INTEGRATION_NAME} attachment of incident {args.get('number', None)}",
+                                      readable_attachment,
+                                      headers=headers,
+                                      removeNull=True)
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix=f'{INTEGRATION_NAME}.attachment',
+        outputs_key_field='id',
+        outputs=response
+    )
+
+
+''' FETCH & MIRRORING COMMANDS'''
+
+
+def fetch_incidents(client: Client,
+                    last_run: Dict[str, Any],
+                    demisto_params: Dict[str, Any]) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """Fetches incidents from TOPdesk."""
+
+    first_fetch_datetime = dateparser.parse(demisto_params.get('first_fetch', '3 days'))
```

Comment:
you should use `parse_date_range` from common server python it basically does this for you.

Reply:
![Screen Shot 2021-03-14 at 10 55 18](https://user-images.githubusercontent.com/72339665/111062788-d4d9fb80-84b3-11eb-9f13-ec258b417eba.png)